### ISSUE TITLE
Babel presets for useBabel

### ISSUE THREAD

**Issue by Ciantic** [s-panferov/awesome-typescript-loader#113]
Right now only way I can target ES6 is like this:

```
{ test: /\.tsx?$/, loaders: ['react-hot', 'babel?presets[]=es2015-loose', 'awesome-typescript-loader?tsconfig=tsconfig.webpack.json'], include: path.resolve(__dirname, "App") },
```

And I probably loose some awesomeness because I can't use "useBabel" option, would it be possible to add presets to the useBabel?

**Comment by Ciantic** [comment]
@ @

**Comment by Ciantic** [comment]
Well. I closed this accidentally.

**Comment by s-panferov** [comment]
@Ciantic 	currently you can just create [`.babelrc` file](https://babeljs.io/docs/usage/babelrc/) with babel settings to the root of your project. I will consider to add babel options to the loader settings, but `.babelrc` is far more universal solution.

**Comment by s-panferov** [comment]
@Ciantic 	`babelOptions` option was added in `v0.17.0-rc.3`, but I would recommend to use `.babelrc` instead.

**Comment by olegstepura** [comment]
Hi! It's currently unclear from README if `useBabel` makes `awesome-typescript-loader` to apply babel, or if I still need to use chained loaders in webpack. I suggest you to make this clear in README. 

So for me it's unclear if I should use this:
```
      {
        test: /\.tsx?$/,
        loaders: ['awesome-typescript'],
        include: path.join(__dirname, srcDir)
      }
```
or this:
```
      {
        test: /\.tsx?$/,
        loaders: ['babel', 'awesome-typescript'],
        include: path.join(__dirname, srcDir)
      }
```
if I want to make `typescript -> ES6` with Typescript and later `ES6 -> ECMA2015` with Babel.
I want this to use generators in Typescript.

Thanks!

**Comment by kesarion** [comment]
This is made even more confusing by the fact that even if you don't have `babel-core` installed and "useBabel" is set to true, it still works just fine...  there's no confirmation whether the output .js files are being transpiled or not. Could we get some clarification about exactly what is required for babel to work with the loader?

**Comment by Ciantic** [comment]
@kesarion nope that is incorrect, now it won't use the cache and else.

this is the way with awesome-typescript-loader

```{ test: /\.tsx?$/, loaders: ['awesome-typescript-loader?{ useBabel: true, babelOptions: { presets: ["es2015-native-modules"], sourceMaps: true } }'], include: appDir },
```

Tested with webpack2

**Comment by kesarion** [comment]
@Ciantic 

Hmm, so I shouldn't use babel-loader after awesome-typescript-loader? Is it used automatically when useBabel is true?

Also, would that configuration work the same way if written in your tsconfig file? For example, like this:

```json
{
    "awesomeTypescriptLoaderOptions": {
        "useBabel": true,
        "babelOptions": {
            "presets": ["es2015-native-modules"],
            "sourceMaps": true
        },
        "useCache": true
    }
}
```

**Comment by Ciantic** [comment]
@kesarion Btw if you just update to typescript@next you can get rid of Babel too since it supports async/await when transpiling to es5. (It does have some bugs though still)

**Comment by kesarion** [comment]
@Ciantic I've tried typescript@next; it would have been great if it worked, but it doesn't compile my app files, it just starts and ends with no error and no files, no idea why. The same project works with the stable version and 2.1.0 dev up to 2.1.0-dev.20160907 where it stops working. Coincidentally, 2.1.0-dev.20160907 is where async/await support is implemented.

**Comment by izhaknys** [comment]
This?

```
            test: /\.tsx?$/,
            loader: 'awesome-typescript-loader',
            query: {
                useBabel: true,
                useCache: true,
                babelOptions: {
                    presets : [ [ 'es2015', { modules: false } ], 'react' ],
                    plugins: [ 'react-hot-loader/babel' ],
                },

```

**Comment by heatherbooker** [comment]
Yeah, I am still confused by this whole thing. Specifically, how to make sure `at-loader` is paying attention to a `.babelrc`.

1. Does specifying `"useBabel": true` in `awesomeTypescriptLoaderOptions` in `tsconfig.json` work? Is it necessary?
2. Does specifying `'awesome-typescript-loader?useBabel=true&useCache=true'` for `test: /\.tsx?$/` in `webpack.config.js` work? Is it necessary?

I just want to know that at-loader is adhering to the specifications of my `.babelrc`.

Thanks! :)
